pkg/signup/service/signup_service.go

```diff
@@ -381,11 +382,9 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, cl namespaced.Client, userna
 		return nil, nil
 	} else if completeCondition.Reason == toolchainv1alpha1.UserSignupUserBannedReason {
 		log.Info(nil, fmt.Sprintf("usersignup: %s is banned", userSignup.GetName()))
-		// UserSignup is banned, let's return a pending approval reason to the client.
-		signupResponse.Status = signup.Status{
-			Reason: toolchainv1alpha1.UserSignupPendingApprovalReason,
-		}
-		return signupResponse, nil
+		// UserSignup is banned, let's return a forbidden error
+		return nil, apierrors.NewForbidden(schema.GroupResource{}, "",
+			errs.New("The account has been banned due to detected abusive activity or suspicious indicators."))
```

Comment:
```suggestion
			errs.New("The account has been suspended for detected abusive or suspicious activity."))
```

You could also consider moving the message string to a const :)

Reply:
yeah, why not - moved to const in cafb76e